hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecordMerger.java

```diff
@@ -56,7 +76,8 @@ public HoodieRecordType getRecordType() {
   private Option<IndexedRecord> combineAndGetUpdateValue(HoodieRecord older, HoodieRecord newer, Schema schema, Properties props) throws IOException {
     Option<IndexedRecord> previousAvroData = older.toIndexedRecord(schema, props).map(HoodieAvroIndexedRecord::getData);
     if (!previousAvroData.isPresent()) {
-      return Option.empty();
+      Option<IndexedRecord> newData = newer.toIndexedRecord(schema, props).map(HoodieAvroIndexedRecord::getData);
+      return newData;
```

Comment:
Is this a bug before?

Reply:
@yihua, @codope, based on the old logic, when the old record is not valid, the merge result is empty, then the merge logic in MergeHandle will try to keep the old data, that is, the invalid old data will be written to the disk again. So I updated it as: when the old data is invalid, but the new one is, we should output the new data; the mergeHandle will check if the new data is a delete or not.